packages/x-charts/src/ChartsBrushOverlay/ChartsBrushOverlay.tsx

```diff
@@ -0,0 +1,117 @@
+'use client';
+import * as React from 'react';
+
+import clsx from 'clsx';
+import { useTheme } from '@mui/material/styles';
+
+import { brushOverlayClasses } from './ChartsBrushOverlay.classes';
+import { selectorChartDrawingArea } from '../internals/plugins/corePlugins/useChartDimensions';
+import {
+  type UseChartBrushSignature,
+  selectorBrushStartX,
+  selectorBrushStartY,
+  selectorBrushCurrentX,
+  selectorBrushCurrentY,
+  selectorBrushConfig,
+} from '../internals/plugins/featurePlugins/useChartBrush';
+import { useSelector } from '../internals/store/useSelector';
+import { useStore } from '../internals/store/useStore';
+
+function BrushRect(props: React.SVGProps<SVGRectElement>) {
+  return (
+    <rect
+      className={brushOverlayClasses.rect}
+      strokeWidth={1}
+      fillOpacity={0.2}
+      pointerEvents={'none'}
+      {...props}
+    />
+  );
+}
+
+export interface ChartsBrushOverlayProps {}
+
+/**
+ * Component that renders visual feedback during brush interaction
+ */
+export function ChartsBrushOverlay(props: ChartsBrushOverlayProps) {
+  const store = useStore<[UseChartBrushSignature]>();
+  const drawingArea = useSelector(store, selectorChartDrawingArea);
+
+  const theme = useTheme();
+
+  const brushStartX = useSelector(store, selectorBrushStartX);
+  const brushStartY = useSelector(store, selectorBrushStartY);
+  const brushCurrentX = useSelector(store, selectorBrushCurrentX);
+  const brushCurrentY = useSelector(store, selectorBrushCurrentY);
+  const brushConfig = useSelector(store, selectorBrushConfig);
+
+  if (
+    brushStartX === null ||
+    brushStartY === null ||
+    brushCurrentX === null ||
+    brushCurrentY === null
+  ) {
+    return null;
+  }
+
+  const { left, top, width, height } = drawingArea;
+
+  // Clamp coordinates to drawing area
+  const clampX = (x: number) => Math.max(left, Math.min(left + width, x));
+  const clampY = (y: number) => Math.max(top, Math.min(top + height, y));
+
+  const startX = clampX(brushStartX);
+  const startY = clampY(brushStartY);
+  const currentX = clampX(brushCurrentX);
+  const currentY = clampY(brushCurrentY);
+  const rectColor = theme.palette.mode === 'light' ? 'black' : 'white';
+
+  // For scatter charts, show only the rectangle without guide lines
+  if (brushConfig === 'xy') {
+    const rectWidth = currentX - startX;
+    const rectHeight = currentY - startY;
+
+    return (
+      <g className={clsx(brushOverlayClasses.root, brushOverlayClasses.x, brushOverlayClasses.y)}>
```

Comment:
I wonder if we should have a `xy` class for this case instead of applying both the `x` and `y` classes. I fear users might want to style the `x`, `y` and `xy` differently. 

What do you think?

Reply:
I had it there before, but they can do an "and" by just `.class-x.class-y`

Reply:
Then the styles of both `x` and `y` would apply, unless they override them in `.class-x.class-y`. 

Not the best DX I think, but there's a workaround, so not a blocker for me

Reply:
still `.class-x:not(.class-y)` would work